### ISSUE TITLE
Check Figure 19 residual plot, make sure in MCMC context

### ISSUE THREAD

**Issue by cgrandin** [pacific-hake/hake-assessment#780]
Written by @andrew-edwards in another issue:

Residual plot Figure 19 - is it already MCMC? I'm going down a rabbit hole and I suspect some else will have an answer. Basically `make_age_comp_pearson_plot()` uses `model$agedbase`, but I can't find whether `...$agedbase` is MLE or MCMC median. 

From the README I see
```
model$extra.mcmc$agedbase$Exp         - median of posterior for expected value for age comps
model$extra.mcmc$agedbase$Pearson     - median of posterior for pearson residuals for age comps
```
and tried both of these in `make_age_comp_pearson_plot()` but no joy. Could make a separate issue but thought someone may know off the top of their head.

**Comment by andrew-edwards** [comment]
@kellijohnson-NOAA asked what errors I was getting. Things like:
```
Error in UseMethod("filter_") : 
  no applicable method for 'filter_' applied to an object of class "c('double', 'numeric')"
```

**Comment by kellijohnson-NOAA** [comment]
when running
```
model$extra.mcmc$agedbase %>% filter(Fleet == subplot)
```

**Comment by andrew-edwards** [comment]
Sorry, when doing
```
make_age_comp_pearson_plot(base.model, subplot = 1, legend.position = "top", alpha = 0.7)
```
Also got 
```
Error in UseMethod("filter_") : 
  no applicable method for 'filter_' applied to an object of class "NULL"
```
from same command while I was playing the function. Don't think this helps you much!

**Comment by kellijohnson-NOAA** [comment]
can't we just change to using base R to filter if that is causing the problem?

**Comment by cgrandin** [comment]
That means you're trying to filter a NULL object, so you have to debug the function

**Comment by cgrandin** [comment]
For example, this works:

```
base.model$extra.mcmc$agedbase %>% filter(Fleet == 1)
```

**Comment by cgrandin** [comment]
Done ff41de664092cddf3f5f0f5cac0f24c3dc2beddd

**Comment by andrew-edwards** [comment]
Everything builds (Sunday morning) for me except Figure 18 (the coloured histograms of age comps):

![image](https://user-images.githubusercontent.com/6977190/107154111-fd575e80-6925-11eb-8037-754f40751c0d.png)

I took a quick look but I think it's related to this issue. Expect Chris can fix it more efficiently than me (thanks!).

**Comment by andrew-edwards** [comment]
I'm guessing the function needs something like
```
extra_mcmc$comp_fishery_median <- extra_mcmc$comp_fishery %>%
    select(Yr, Age, Obs = Obs_med, Exp = Exp_med, Pearson = Pearson_med)
```
from your commit above.

**Comment by kellijohnson-NOAA** [comment]
needs Exp_lower and Exp_upper as well

**Comment by cgrandin** [comment]
This will be resolved by downloading the base and age1 RDS files one final time. They contain the objects I worked on last night which build that figure. I just finished uploading them.

**Comment by andrew-edwards** [comment]
Mine built. Checking everyone does, then can close:
- [ ] Kelli
- [ ] Aaron
- [x] Andy

**Comment by aaronmberger-nwfsc** [comment]
I checked the box here because my Figure 18 built (colored age comp fits) but my Figure 19 (pearson residuls) still has a caption that make me think this isn't done yet?  But I think Chris said done above?  So I'm a bit confused and not sure if I should be checking this box or not?

**Comment by cgrandin** [comment]
I just fixed that caption and yes it is done! b00bf3b8e9c2356aa902f7f84c2e081377a35219
